### PR TITLE
[kots]: add component customization docs

### PR DESCRIPTION
**Pull request by MrSimonEmms**
> **IMPORTANT** - not to be released before July release as the underlying feature is not made public until then

## Description

Add component customization docs with proxy service type

## Related Issue(s)

Fixes https://github.com/gitpod-io/gitpod/issues/11011

## How to test

Follow docs

## Release Notes

```release-note
[kots]: add component customization docs
```

## Documentation

<a href="https://gitpod-staging.com/#https://github.com/gitpod-io/website/pull/2332"></a>

<a href="https://gitpod.io/#https://github.com/gitpod-io/website/pull/2332"></a>

### CODE REVIEW

File: gitpod/docs/self-hosted/latest/advanced/components.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+# Component Customization
+
+In this guide, we expect you to have a cluster up and running that [meets the requirements](../cluster-set-up) and has the [required components](../required-components) installed and configured (at least cert-manager is needed).
+
+This guide assumes that you are using our default installation method from our [getting started guide](../getting-started).
+
+## Rationale
+
+Gitpod self-hosted is built to work by default, but it also has to be incredibly flexible. Our installation defaults will work for the vast majority of users, but there will be occassions when it does not.
```
Comment (lucasvaltl) [inline gitpod/docs/self-hosted/latest/advanced/components.md:18]: I'dd add in here at the end "and in order to add more flexibility, you have the ability to customise some of Gitpod's components."

File: gitpod/docs/self-hosted/latest/advanced/components.md [hunk 2]
```diff
@@ -0,0 +1,28 @@
+---
+section: self-hosted/latest
```
Comment (lucasvaltl) [inline gitpod/docs/self-hosted/latest/advanced/components.md:2]: We should add the following in here under `section`:

`subsection: installation-guides`

--> This makes the second level hierarchy in the menu not collapse.

**Review by lucasvaltl — APPROVED** [review]
LGTM - some comments I'd like to see addressed, but happy for you to merge without another review. 

Note - I shipped a [change](https://github.com/gitpod-io/website/pull/2337) to the structure of our docs. I don't think this impacts this PR but it might.

**Comment by MrSimonEmms** [comment]
@lucasvaltl I've resolved both of those comments. We'll publish this when we do the July release

**Pull request merged**